frontend/src/lib/WidgetStateManager.ts

```diff
@@ -193,6 +194,32 @@ export class WidgetStateManager {
     })
   }
 
+  public sendUploadFileMessage(
+    id: string,
+    name: string,
+    lastModified: number,
+    data: Uint8Array
+  ): void {
+    const limit = 10 * 1e6 //10MB
+
+    const newFileMessage: NewFile = new NewFile()
+    newFileMessage.id = id
+    newFileMessage.name = name
+    newFileMessage.size = data.length
+    newFileMessage.lastModified = lastModified
+    newFileMessage.chunks = Math.ceil(data.length / limit)
+    this.sendBackMsg({ newFile: newFileMessage })
+
+    for (let i = 0; i < newFileMessage.chunks; i++) {
+      const message: FileChunk = new FileChunk()
```

Comment:
nit: `const message = new FileChunk()` (actually I think our ESlint should be catching/fixing these? maybe it's not running properly?)